marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs

```diff
@@ -102,22 +103,53 @@ class Flow {
     if (cachedFlows[name] === undefined) {
       let uriMatches = cts.uriMatch('/flows/' + name + '.flow.json', ['case-insensitive'], cts.directoryQuery("/flows/"));
       // cache flow to prevent repeated calls.
+      let flowObject;
       if (fn.count(uriMatches) === 1) {
-        cachedFlows[name] = cts.doc(fn.head(uriMatches)).toObject();
+        flowObject = cts.doc(fn.head(uriMatches)).toObject();
       } else if (fn.count(uriMatches) > 1) {
         for (let uri of uriMatches) {
           if (uri === '/flows/' + name + '.flow.json') {
-            cachedFlows[name] = cts.doc(uriMatches).toObject();
+            flowObject  = cts.doc(uriMatches).toObject();
             break;
           }
         }
       } else {
         cachedFlows[name] = null;
+        return null;
       }
+      flowObject = this.addSettingsToFlow(name, flowObject);
+      cachedFlows[name] = flowObject;
     }
     return cachedFlows[name];
   }
 
+  addSettingsToFlow(flowName, flowNode){
```

Comment:
1. WriteStepRunner requires the "fully populated" document, I believe steps using QueryStepRunner don't
2. I initially changed core.sjs/getArtifact() to return "fully populated" flow doc  (FlowManager.getFlow() and all clients use it).  It worked fine but whenever a flow is edited (add/remove step) , the flows that gets written is the fully populated document (since creating a new step, gets a flow (fully populated doc) and adds steps to it and persists it. So, I moved this method from core.sjs 's getArtifact to impl/flows.sjs 's getFlow() method

Reply:
Added a new endpoint that gets fullFlow() to be used by FlowRunner